```
_________________ ERROR collecting tests/system/test_to_gbq.py _________________
[tests/system/test_to_gbq.py:392](https://cs.corp.google.com/piper///depot/google3/tests/system/test_to_gbq.py?l=392): in <module>
    dtype=pandas.ArrowDtype(
[.nox/system-3-8/lib/python3.8/site-packages/pandas/__init__.py:258](https://cs.corp.google.com/piper///depot/google3/.nox/system-3-8/lib/python3.8/site-packages/pandas/__init__.py?l=258): in __getattr__
    raise AttributeError(f"module 'pandas' has no attribute '{name}'")
E   AttributeError: module 'pandas' has no attribute 'ArrowDtype'
```

Looks like I need to update the tests to be compatible with older pandas.

--- a/setup.py
+++ b/setup.py
@@ -25,7 +25,7 @@ dependencies = [
     "db-dtypes >=1.0.4,<2.0.0",
     "numpy >=1.18.1",
     "pandas >=1.1.4",
-    "pyarrow >=3.0.0",
+    "pyarrow >=4.0.0",
     "pydata-google-auth >=1.5.0",
     # Note: google-api-core and google-auth are also included via transitive
     # dependency on google-cloud-bigquery, but this library also uses them

--- a/testing/constraints-3.8.txt
+++ b/testing/constraints-3.8.txt
@@ -15,7 +15,7 @@ google-cloud-bigquery==3.4.2
 google-cloud-bigquery-storage==2.16.2
 numpy==1.18.1
 pandas==1.1.4
-pyarrow==3.0.0
+pyarrow==4.0.0
 pydata-google-auth==1.5.0
 Shapely==1.8.4
 tqdm==4.23.0

--- a/tests/system/test_to_gbq.py
+++ b/tests/system/test_to_gbq.py
@@ -396,13 +396,17 @@ DATAFRAME_ROUND_TRIPS = [
                                     ("project", pyarrow.string()),
                                 ]
                             )
-                        ),
+                        )
+                        if hasattr(pandas, "ArrowDtype")
+                        else "object",
                     ),
                     "arrow_array": pandas.Series(
                         [[1, 2, 3], None, [4, 5, 6]],
                         dtype=pandas.ArrowDtype(
                             pyarrow.list_(pyarrow.int64()),
-                        ),
+                        )
+                        if hasattr(pandas, "ArrowDtype")
+                        else "object",
                     ),
                     "arrow_array_of_struct": pandas.Series(
                         [
@@ -412,7 +416,9 @@ DATAFRAME_ROUND_TRIPS = [
                         ],
                         dtype=pandas.ArrowDtype(
                             pyarrow.list_(pyarrow.struct([("test", pyarrow.string())])),
-                        ),
+                        )
+                        if hasattr(pandas, "ArrowDtype")
+                        else "object",
                     ),
                 },
             ),
@@ -423,20 +429,26 @@ DATAFRAME_ROUND_TRIPS = [
                         [{"test": "str1"}, {"test": "str2"}, {"test": "str3"}],
                         dtype=pandas.ArrowDtype(
                             pyarrow.struct([("test", pyarrow.string())]),
-                        ),
+                        )
+                        if hasattr(pandas, "ArrowDtype")
+                        else "object",
                     ),
                     # Array of DATETIME requires inspection into list elements.
                     # See:
                     # https://github.com/googleapis/python-bigquery/pull/1061
                     "object_array_datetime": pandas.Series(
                         [[], [datetime.datetime(1998, 9, 4, 12, 0, 0)], []],
-                        dtype=pandas.ArrowDtype(pyarrow.list_(pyarrow.timestamp("us"))),
+                        dtype=pandas.ArrowDtype(pyarrow.list_(pyarrow.timestamp("us")))
+                        if hasattr(pandas, "ArrowDtype")
+                        else "object",
                     ),
                     "object_array_of_struct": pandas.Series(
                         [[], [{"test": "str4"}], []],
                         dtype=pandas.ArrowDtype(
                             pyarrow.list_(pyarrow.struct([("test", pyarrow.string())])),
-                        ),
+                        )
+                        if hasattr(pandas, "ArrowDtype")
+                        else "object",
                     ),
                     "arrow_struct": pandas.Series(
                         [
@@ -451,19 +463,25 @@ DATAFRAME_ROUND_TRIPS = [
                                     ("project", pyarrow.string()),
                                 ]
                             )
-                        ),
+                        )
+                        if hasattr(pandas, "ArrowDtype")
+                        else "object",
                     ),
                     "arrow_array": pandas.Series(
                         [[1, 2, 3], [], [4, 5, 6]],
                         dtype=pandas.ArrowDtype(
                             pyarrow.list_(pyarrow.int64()),
-                        ),
+                        )
+                        if hasattr(pandas, "ArrowDtype")
+                        else "object",
                     ),
                     "arrow_array_of_struct": pandas.Series(
                         [[{"test": "str5"}], [], [{"test": "str6"}, {"test": "str7"}]],
                         dtype=pandas.ArrowDtype(
                             pyarrow.list_(pyarrow.struct([("test", pyarrow.string())])),
-                        ),
+                        )
+                        if hasattr(pandas, "ArrowDtype")
+                        else "object",
                     ),
                 },
             ),

--- a/tests/unit/schema/test_pandas_to_bigquery.py
+++ b/tests/unit/schema/test_pandas_to_bigquery.py
@@ -6,9 +6,9 @@ import collections
 import datetime
 import operator
 
+from google.cloud.bigquery import schema
 import pandas
 import pytest
-from google.cloud.bigquery import schema
 
 
 @pytest.fixture

